www/install/step_upgrade/step3.php

```diff
@@ -43,55 +43,59 @@
 require_once '../steps/functions.php';
 $template = getTemplate('templates');
 
+require_once realpath(dirname(__FILE__) . "/../../../config/centreon.config.php");
+include_once _CENTREON_PATH_ . "/www/class/centreonDB.class.php";
+
 $title = _('Release notes');
 
-if (is_file('../RELEASENOTES.html')) {
-    $contents = "<div id='releasenotes'></div>";
-} else {
-    $releasenotesContent = '';
-    if (file_exists('../RELEASENOTES')) {
-        $releasenotesContent = file_get_contents('../RELEASENOTES');
+$db = new CentreonDB();
+$res = $db->query("SELECT `value` FROM `informations` WHERE `key` = 'version'");
+$row = $res->fetchRow();
+$current = $row['value'];
+
+$_SESSION['CURRENT_VERSION'] = $current;
+
+/*
+** To find the final version that we should update to, we will look in
+** the www/install/php directory where all PHP update scripts are
+** stored. We will extract the target version from the filename and find
+** the farthest version to the current version.
+*/
+$next = '';
+if ($handle = opendir('../php')) {
+    while (false !== ($file = readdir($handle))) {
+        if (preg_match('/Update-([a-zA-Z0-9\-\.]+)\.php/', $file, $matches)) {
```

Comment:
```suggestion
        if (preg_match('/Update-(\d+\.\d+\.\d+)\.php/', $file, $matches)) {
```

Reply:
This will not work with versions like "20.04.0-rc2" no ?

Reply:
indeed